js/noms/src/database.js

```diff
@@ -78,7 +65,24 @@ export default class Database {
     return true;
   }
 
-  async commit(datasetId: string, commit: Commit<any>): Promise<Database> {
+  // Commit updates the commit that ds points at. If parents is provided then the promise
+  // is rejected if the commit does not descend from the parents.
+  async commit(ds: Dataset, v: Value, parents: ?Array<Ref<Commit<any>>> = undefined):
+  Promise<Dataset> {
+    if (!parents) {
+      const headRef = await ds.headRef();
+      parents = headRef ? [headRef] : [];
+    }
+    const commit = new Commit(v, new Set(parents));
+    try {
+      const commitRef = await this._doCommit(ds.id, commit);
+      return new Dataset(this, ds.id, Promise.resolve(commitRef));
```

Comment:
OK. I like what you had before better. If we are using async functions we should minimize the number of `then` usages.